pkgdb/src/buildenv/realise.cc

```diff
@@ -721,51 +725,65 @@ makeActivationScripts( nix::EvalState & state, resolver::Lockfile & lockfile )
               indexOfQuoteChar += 4;
             }
 
-          commonActivate << nix::fmt( "export %s='%s'\n", name, value );
+          bashScript << nix::fmt( "export %s='%s'\n", name, value );
+          zshScript << nix::fmt( "export %s='%s'\n", name, value );
+        }
+    }
+
+  /* Add profile scripts */
+  auto profile = manifest.profile;
+  if ( profile.has_value() )
+    {
+      if ( profile->common.has_value() )
+        {
+          debugLog( "adding 'profile.common' to activation scripts" );
+          addScriptToScriptsDir( profile->common.value(),
+                                 tempDir,
+                                 "profile-common" );
+          appendSourcedScript( "profile-common", bashScript );
+          appendSourcedScript( "profile-common", zshScript );
+        }
+      if ( profile->bash.has_value() )
+        {
+          debugLog( "adding 'profile.bash' to activation scripts" );
+          addScriptToScriptsDir( profile->bash.value(),
+                                 tempDir,
+                                 "profile-bash" );
+          appendSourcedScript( "profile-bash", bashScript );
+        }
+      if ( profile->zsh.has_value() )
+        {
+          debugLog( "adding 'profile.zsh' to activation scripts" );
+          addScriptToScriptsDir( profile->zsh.value(), tempDir, "profile-zsh" );
+          appendSourcedScript( "profile-zsh", zshScript );
         }
     }
 
   /* Add 'on-activate' script. */
-  auto hook = lockfile.getManifest().getManifestRaw().hook;
+  auto hook = manifest.hook;
   if ( hook.has_value() )
     {
       if ( hook->script.has_value() )
         {
           debugLog( "adding 'hook.script' to activation scripts" );
-          addScriptToScriptsDir( hook->script.value(),
-                                 tempDir,
-                                 "hook.sh",
-                                 commonActivate,
-                                 true );
+          addScriptToScriptsDir( hook->script.value(), tempDir, "hook-script" );
+          appendSourcedScript( "hook-script", bashScript );
+          appendSourcedScript( "hook-script", zshScript );
```

Comment:
question (nb): is doing the backwards compatibility with `hook.script`?
Question : is this writing both bash and zsh scripts into the same file?

Reply:
Yes, it is handling the deprecated `hook.script`.  Adding a comment as such.

It is writing one script file, then adding a `source xxx` of that file to both the bash and zsh activation scripts.

Also adjusting the logic, since I think we should support both `hook.script` and `hook.on-activate`.  The latter is the only current way to have a `bash` executed activation script.  I don't think they conflict.